### DISCUSSION TITLE
Localhost with port number does not work on Mac

### DISCUSSION THREAD

**Discussion by ghost** [datenstrom/community#129] [Report a bug] [answered]
Hi,
Looks interesting!
How do I get the Admin to play with localhost ( http://localhost:8888/yellow/ ) !?
Disable https/ssl !? How do I do that ?

**Comment by markseuffert** [comment]
Hi, 

let me see how I can help. You don't have to disable anything. To add a user account, open your website in a web browser and click on "**edit this page**". What happens when you do this?

**Comment by ghost** [comment]
"Unable to connect" !

**Comment by ghost** [comment]
Admin works when I upload it to my remote live server but not on localhost dev. (mac mamp)

**Comment by markseuffert** [comment]
Can you add `[debug]` and `[debug server]` to a page and show the settings? Thanks!

**Comment by ghost** [comment]
When I hit "edit" the url changes and stripping out the "8888" to "http://localhost/yellow/edit/"
I have tried the following:
Rewritebase / yellow (.htaccess)
ServerName: http://localhost:8888 (config.ini)

still showing "server not found" / or "unable to connect"

**Comment by ghost** [comment]
uhm how do I turn on debug mode ?
Anyway ADMIN is not working with mamp for mac on localhost, the rest of the site is of course working !

**Comment by ghost** [comment]
I haven´t touch the config.ini, and I created a user through the terminal.. as instructed.

**Comment by markseuffert** [comment]
Sorry for the trouble, let's figure out where the error is located with some debugging information. 

Please go to your Yellow folder. Open file `content/1-home/page.txt` and edit it like below:

```
---
Title: Home
---
Your website works! 

You can now [edit this page] or use your text editor.  
For more information see [Yellow documentation](https://github.com/datenstrom/yellow/wiki).

[debug]

[debug server]
```

This will show current software version and server settings.

**Comment by ghost** [comment]
Yellow 0.6.1, PHP 5.6.2, Apache/2.2.29 Darwin

ServerScheme: http
ServerName: localhost
ServerBase: /yellow
ServerTime: Europe/Berlin

**Comment by markseuffert** [comment]
Oh... so the port number is missing, looks like a bug. Let me try to reproduce it on my computer.

**Comment by ghost** [comment]
hmm serverName should be "localhost:8888" maybe?
because the site is running on http://localhost:8888/yellow/

**Comment by ghost** [comment]
if I change the settings in config.ini

 Yellow 0.6.1, PHP 5.6.2, Apache/2.2.29 Darwin

ServerScheme: http
ServerName: localhost:8888
ServerBase: /yellow
ServerTime: Europe/Berlin

and hit "edit this page" I get >

The page isn't redirecting properly
Firefox has detected that the server is redirecting the request for this address in a way that will never complete.
    This problem can sometimes be caused by disabling or refusing to accept cookies.

**Comment by markseuffert** [comment]
I can confirm that it's a bug. Any web server running non-standart port numbers is affected, the problem is in function getServerName() which doesn't detect the correct server name.

I plan to make an update this week. Are you available for testing?

**Comment by ghost** [comment]
```
function getServerName()
    {
        return $_SERVER['HTTP_HOST'];
        //return $_SERVER["SERVER_NAME"];
    }
```

**Comment by ghost** [comment]
bah I just freeeaking changed the port number to : 80 instead of : 8888, so now I can have http://localhost/ instead of http://localhost:8888/

ISSUE RESOLVED !!

Thanks for yellow and your support !!

**Comment by markseuffert** [comment, marked as answer]
Should be working now.
